crate_universe/BUILD.bazel

```diff
@@ -86,6 +86,12 @@ rust_test(
         "@rules_rust//rust/toolchain:current_rustc_files",
         "//crate_universe/test_data/serialized_configs",
     ],
+    env = {
+        "CARGO": "$(rootpath @rules_rust//rust/toolchain:current_cargo_files)",
```

Comment:
What tests need cargo and rustc? It'd be ideal if tests didn't need to invoke those binaries.

Reply:
`feature_generator` test in `metadata.rs` -- this test covers the whole flow which takes a spliced workspace (w/ Cargo.lock) and runs `cargo tree` for each desired platform triplet, and then parses the output out into a feature map.

Especially since `cargo tree` output isn't guaranteed to be table (and isn't intended for parsing), I think this is a useful test case to have. I can perhaps factor it out into a different test (more like an integration test)?  I'll be a little sad, but I can also just delete it if you'd think that's best... WDYT?

Reply:
I think https://github.com/bazelbuild/rules_rust/pull/1710#discussion_r1073076313 kinda encompasses this thread as well. My hesitation to adding cargo to this test is it becomes easier for folks to call it and I'd really like to keep all invocations to cargo inside the splicing phase and I'd want the tests to be the same. The need to invoke cargo definitely seems like an integration test, which I think could be Ok but then I'd want to test things like "if you call cargo tree on this workspace with these flags, you get this deterministic output". That way tests would be written around that output metadata and we'd be able to catch regressions.

Reply:
I refactored this as (somewhat of) an integration test from "splicing input" to output metadata. Hopefully it's somewhat more in line with what you're looking for.